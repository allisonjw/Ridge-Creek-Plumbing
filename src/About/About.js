import React from 'react';
import './About.scss';
import PNW from '../images/PNW_1.jpeg';
import commercial from '../images/commercial_plumbing.jpeg';
import boots from '../images/boots.jpeg';
import plumbing from '../images/plumbing_1.jpeg';
import Fade from 'react-reveal/Fade';

export const About = () => {
    return (
      <main>
        <div className="about_banner" aria-label="slideshow of images">
          <figure>
            <div className="about_img" style={{ backgroundImage:`linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(${PNW})` }}> 
            </div>
              <figcaption>
                Image of the puget sound in the Pacific Northwest. 
              </figcaption>
            </figure>
            <figure>
              <div className="about_img" style={{ backgroundImage:`linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(${commercial})` }}> 
              </div>
            <figcaption>
                One of our commercial plumbing projects. 
            </figcaption>
            </figure>
            <figure>
              <div className="about_img" style={{ backgroundImage:`linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(${boots})` }}> 
              </div>
            <figcaption>
                We work rain or shine.
            </figcaption>
            </figure>
            <figure>
              <div className="about_img" style={{ backgroundImage:`linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(${plumbing})` }}> 
              </div>
            <figcaption>
                One of our residential jobs in progress.
            </figcaption>
            </figure>
        </div>
        <Fade bottom>
        <section className="section_about-why">
            <h2 className="about_h2-why">Why Ridgecreek</h2>
            <p className="about_paragraph">
              Whether working with a home owner to replace a plumbing appliance, a landlord to make tenant improvements, or a commercial contractor to install a building’s plumbing and hydronic system, our commitment to quality of service and workmanship is why we are trusted across the South Sound region.

              We are a signatory with United Association Local 26, Plumbers and Pipefitters Union, and we are also a member of the Olympia Master Builders Association. We pride ourselves on our strong relationships with local contractors and suppliers ensuring we are always able to deliver the highest level of service to our clients.
            </p>
        </section>
        </Fade>
        <Fade bottom>
        <section className="section_about-history">
            <h2 className="about_h2">Our History</h2>
            <p className="about_paragraph">
              Ridgecreek Plumbing specializes in residential and commercial plumbing installation, water and gas line repair, re-pipe, remodel, and new construction, serving Olympia, Lacey, Tumwater, and the greater South Sound region.

              Established in 2009, Ridgecreek Plumbing was founded by Garth Magaro, a Journeyman Plumber with nearly two decades experience in commercial and residential work serving corporate, academic, government, and residential customers. With a focus on excellent customer service, safety and code compliance, and a fine attention to detail, we ensure that each job exceeds customer expectations with a dedication to the highest standards of quality.
            </p>
        </section>
        </Fade>
        <Fade bottom>
        <section className="section_about-mission">
            <h2 className="about_h2">Mission Statement</h2>
            <p className="about_paragraph">
              We are interested in learning about your plumbing and hydronic projects. Our goal is to be your reliable partner in projects of any scope and size. We look forward to working with you.
            </p>
        </section>
        </Fade>
        <aside className="aside_testimonial">
          <h2 className="about_h2">Testimonials</h2>
          <p className="testimonials_description">Our focus is meeting your goals and providing the best customer service possible. See what some of our valued clients are saying about us.</p>
        </aside>
        <section className="section_about-testimonials">
          <input id="go1" type="radio" name="select" className="input" checked/>
          <input id="go2" type="radio" name="select" className="input"checked/>
          <input id="go3" type="radio" name="select" className="input" checked/>
          <input id="go4" type="radio" name="select" className="input"checked/>
	     <figure className="about_testimonials">
		   <figure className="testimonial" id="p1">
			<p className="testimonial_content">"Garth and his team at RidgeCreek has been there for me multiple times for projects (and one homeowner caused emergency) and I’ve been incredibly impressed with their responsiveness and attention to detail. Although they work on large commercial projects, they treated mine like it was just as important.  I would highly recommend them for any plumbing project."<br></br>~ Chris Olympia, WA</p>
			<label for="go2" className="btn btn-inside next">›</label>
		  </figure>
		  <figure className="testimonial" id="p2">
			<p className="testimonial_content">"lorem ipsum testimonial two"</p>
			<label for="go1" className="btn btn-inside prev">‹</label>
			<label for="go3" className="btn btn-inside next">›</label>
		  </figure>
		  <figure className="testimonial" id="p3">
			<p className="testimonial_content">"Lorem ipsum testimonial 3"</p>
			<label for="go2" className="btn btn-inside prev">‹</label>
			<label for="go4" className="btn btn-inside next">›</label>
		  </figure>
          <figure className="testimonial" id="p4">
			<p className="testimonial_content">"lorem ipsum testimonial 4"</p>
			<label for="go3" className="btn btn-inside prev">‹</label>
			<label for="go1" className="btn btn-inside next">«</label>
		  </figure>
	    </figure>
          <label for="go1" className="btn btn-outside go1">1</label>
          <label for="go2" className="btn btn-outside go2">2</label>
          <label for="go3" className="btn btn-outside go3">3</label>
          <label for="go4" className="btn btn-outside go4">4</label>
        </section>
      </main>
    )
};

export default About;