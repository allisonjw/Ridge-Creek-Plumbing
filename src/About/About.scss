$image-margin: 1em;

main {
    animation: fade-in 1s ease-out both 0.3s;
}

@keyframes fade-in {
    from {opacity: 0%;}
    to {opacity: 100%;}
}

.about_banner {
    width: 100%;
    height: 300px;
    background-color: pink;
    margin: $image-margin auto;
    position: relative;
    figure {
      position: absolute;
      opacity: 0;
        &:nth-child(1) {
          animation: xfade 16s 12s infinite;
        }
        &:nth-child(2) {
          animation: xfade 16s 8s infinite;
        }
        &:nth-child(3) {
          animation: xfade 16s 4s infinite;
        }
        &:nth-child(4) {
          animation: xfade 16s 0s infinite;
        }
        img {
          width: 100%;
          height: autoscale;
        }
        figcaption {
          position: absolute;
          top: 0; // align to top of relative parent container
          color: #fff;
          background: rgba(0, 0, 0, 0.3);
          font-size: .8rem;
          padding: 5px 10px;
          opacity: 0;
          transition: opacity .5s; //fade out
        }
    }
  // set hover transition to css slideshow container
  &:hover figure figcaption {
    transition: opacity .5s; //fade in
    opacity: 1;
  }
}


@keyframes xfade{
  0% {opacity: 0;}
  25% {opacity: 1;}
  50% {opacity: 0;}
  100% {opacity: 0;}
}

