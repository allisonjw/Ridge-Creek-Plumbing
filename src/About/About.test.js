import React from 'react';
import About from './About';
import { shallow } from 'enzyme';

describe('About', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<About />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});