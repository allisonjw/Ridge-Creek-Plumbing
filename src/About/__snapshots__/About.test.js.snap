// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match the snapshot 1`] = `
<main>
  <div
    aria-label="slideshow of images"
    className="about_banner"
  >
    <figure>
      <div
        className="about_img"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(PNW_1.jpeg)",
          }
        }
      />
      <figcaption>
        Image of the puget sound in the Pacific Northwest.
      </figcaption>
    </figure>
    <figure>
      <div
        className="about_img"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(commercial_plumbing.jpeg)",
          }
        }
      />
      <figcaption>
        One of our commercial plumbing projects.
      </figcaption>
    </figure>
    <figure>
      <div
        className="about_img"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(boots.jpeg)",
          }
        }
      />
      <figcaption>
        We work rain or shine.
      </figcaption>
    </figure>
    <figure>
      <div
        className="about_img"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(plumbing_1.jpeg)",
          }
        }
      />
      <figcaption>
        One of our residential jobs in progress.
      </figcaption>
    </figure>
  </div>
  <Fade
    bottom={true}
  >
    <section
      className="section_about-why"
    >
      <h2
        className="about_h2"
      >
        Why Ridgecreek
      </h2>
      <hr
        className="about_h2-underline"
      />
      <p
        className="about_paragraph"
      >
        Whether working with a home owner to replace a plumbing appliance, a landlord to make tenant improvements, or a commercial contractor to install a building’s plumbing and hydronic system, our commitment to quality of service and workmanship is why we are trusted across the South Sound region.
        <br />
        <br />
        We are a signatory with United Association Local 26, Plumbers and Pipefitters Union, and we are also a member of the Olympia Master Builders Association. We pride ourselves on our strong relationships with local contractors and suppliers ensuring we are always able to deliver the highest level of service to our clients.
      </p>
    </section>
    <section
      className="section_about-history"
    >
      <h2
        className="about_h2"
      >
        Our History
      </h2>
      <hr
        className="about_h2-underline"
      />
      <p
        className="about_paragraph"
      >
        Established in 2009, Ridgecreek Plumbing was founded by Garth Magaro, a Journeyman Plumber with nearly two decades experience in commercial and residential work serving corporate, academic, government, and residential customers. With a focus on excellent customer service, safety and code compliance, and a fine attention to detail, we ensure that each job exceeds customer expectations with a dedication to the highest standards of quality.
        <br />
        <br />
        One of the things that makes Ridgecreek Plumbing, LLC unique is that Garth's education, training, and experience has helped shape Ridgecreek Plumbing into a business that offers a wide range of mechanical services. From residential service and repair for homeowners, to light commercial and tenant improvement projects for small business owners and landlords, to commercial and industrial projects for general contractors, our plumbers are skilled to do it all.
      </p>
    </section>
    <section
      className="section_about-mission"
    >
      <h2
        className="about_h2"
      >
        Mission Statement
      </h2>
      <hr
        className="about_h2-underline"
      />
      <p
        className="about_paragraph"
      >
        For our homeowner clients, one of the most important things we can offer, at no cost, is reliable information and resources for them to learn about hiring contractors and protecting their rights.  A contractor who is licensed, bonded and insured can often provide a high quality of workmanship and ensure adequate protections for homeowners.
        <br />
        <br />
        Washington State's Department of Labor and Industries is responsible for overseeing contractors and the following links are great resources for homeowners:
        <br />
      </p>
      <ul
        aria-label="container to hold links for customer education"
        className="mission_links-holder"
      >
        <li
          className="about_li"
        >
          This
          <a
            className="about_mission-links"
            href="https://secure.lni.wa.gov/verify/"
            rel="noopener noreferrer"
            target="_blank"
          >
            link
          </a>
          provides a tool for the general public to verify a contractor
        </li>
        <li
          className="about_li"
        >
          This
          <a
            className="about_mission-links"
            href="https://www.lni.wa.gov/licensing-permits/contractors/hiring-a-contractor/protect-my-home"
            rel="noopener noreferrer"
            target="_blank"
          >
            link
          </a>
          provides information about the risks one assumes when hiring someone who is not a contractor
        </li>
        <li
          className="about_li"
        >
          This
          <a
            className="about_mission-links"
            href="https://www.lni.wa.gov/licensing-permits/contractors/hiring-a-contractor/hire-smart-step-by-step"
            rel="noopener noreferrer"
            target="_blank"
          >
            link
          </a>
          provides information about hiring a contractor
        </li>
        <li
          className="about_li"
        >
          This
          <a
            className="about_mission-links"
            href="https://lni.wa.gov/licensing-permits/contractors/problems-with-a-contractor/"
            rel="noopener noreferrer"
            target="_blank"
          >
            link
          </a>
          provides information about what to do if you have problems with a contractor
        </li>
      </ul>
    </section>
  </Fade>
  <aside
    className="aside_testimonial"
  >
    <h2
      className="about_h2"
    >
      Testimonials
    </h2>
    <hr
      className="about_testimonial-underline"
    />
    <p
      className="testimonials_description"
    >
      Our focus is meeting your goals and providing the best customer service possible. See what some of our valued clients are saying about us.
    </p>
  </aside>
  <section
    className="section_about-testimonials"
  >
    <input
      checked={true}
      className="input"
      id="go1"
      name="select"
      type="radio"
    />
    <input
      checked={true}
      className="input"
      id="go2"
      name="select"
      type="radio"
    />
    <input
      checked={true}
      className="input"
      id="go3"
      name="select"
      type="radio"
    />
    <input
      checked={true}
      className="input"
      id="go4"
      name="select"
      type="radio"
    />
    <figure
      className="about_testimonials"
    >
      <figure
        className="testimonial"
        id="p1"
      >
        <p
          className="testimonial_content"
        >
          "Garth and his team at RidgeCreek has been there for me multiple times for projects (and one homeowner caused emergency) and I’ve been incredibly impressed with their responsiveness and attention to detail. Although they work on large commercial projects, they treated mine like it was just as important.  I would highly recommend them for any plumbing project."
          <br />
          ~ Chris Olympia, WA
        </p>
        <label
          className="btn btn-inside next"
          htmlFor="go2"
        >
          ›
        </label>
      </figure>
      <figure
        className="testimonial"
        id="p2"
      >
        <p
          className="testimonial_content"
        >
          "These guys are top quality.  They fixed my faulty pipes and hot water system that caused water damage.  The insurance company selected them for me, and the insurance company doesn't mess around, only select competent people.  They were gentlemen the whole time."
          <br />
          ~ Stan Gig Harbor, WA
        </p>
        <label
          className="btn btn-inside prev"
          htmlFor="go1"
        >
          ‹
        </label>
        <label
          className="btn btn-inside next"
          htmlFor="go3"
        >
          ›
        </label>
      </figure>
      <figure
        className="testimonial"
        id="p3"
      >
        <p
          className="testimonial_content"
        >
          "I have used Ridgecreek a few times. They are my go to Plumber and do fantastic work. The owner, Garth, is a great person to work with.  They are responsive, knowledgable, honest, courteous, and very reasonably priced.  Their workmanship is high quality.  I strongly recommend them."
          <br />
          ~ Lauren, Olympia WA
        </p>
        <label
          className="btn btn-inside prev"
          htmlFor="go2"
        >
          ‹
        </label>
        <label
          className="btn btn-inside next"
          htmlFor="go4"
        >
          ›
        </label>
      </figure>
      <figure
        className="testimonial"
        id="p4"
      >
        <p
          className="testimonial_content"
        >
          "Pellentesque habitant morbi tristique senectus et netus. Faucibus a pellentesque sit amet porttitor eget. Sit amet volutpat consequat mauris nunc congue nisi. Augue lacus viverra vitae congue eu consequat. Libero justo laoreet sit amet cursus sit."
          <br />
          ~ Mary Olympia, WA
        </p>
        <label
          className="btn btn-inside prev"
          htmlFor="go3"
        >
          ‹
        </label>
        <label
          className="btn btn-inside next"
          htmlFor="go1"
        >
          «
        </label>
      </figure>
    </figure>
    <label
      className="btn btn-outside go1"
      htmlFor="go1"
    >
      1
    </label>
    <label
      className="btn btn-outside go2"
      htmlFor="go2"
    >
      2
    </label>
    <label
      className="btn btn-outside go3"
      htmlFor="go3"
    >
      3
    </label>
    <label
      className="btn btn-outside go4"
      htmlFor="go4"
    >
      4
    </label>
  </section>
</main>
`;
